physics/module_sf_mynn.F90

```diff
@@ -659,12 +659,73 @@ SUBROUTINE SFCLAY1D_mynn(                                      &
+                  E1=SVP1*EXP(4648*(1./273.15 - 1./TSK_wat(I)) - &
+                    & 11.64*LOG(273.15/TSK_wat(I)) + 0.02265*(273.15 - TSK_wat(I)))
+               ELSE
+                  !SATURATION VAPOR PRESSURE WRT WATER (Bolton 1980)
+                  E1=SVP1*EXP(SVP2*(TSK_wat(I)-SVPT0)/(TSK_wat(i)-SVP3))
+               ENDIF
+               QSFC_wat(I)=EP2*E1/(PSFC(I)-ep_3*E1)             !specific humidity    
+               QSFCMR_wat(I)=EP2*E1/(PSFC(I)-E1)                !mixing ratio 
+            ENDIF
+            IF (dry(i)) THEN
+              if( lsm == lsm_ruc) then
+                QSFCMR_lnd(I)=QSFC_lnd(I)/(1.-QSFC_lnd(I))       !mixing ratio 
+              else
+               TABS = 0.5*(TSK_lnd(I) + T1D(I))
+               IF (TABS .LT. 273.15) THEN
+                  !SATURATION VAPOR PRESSURE WRT ICE (SVP1=.6112; 10*mb)
+                  E1=SVP1*EXP(4648*(1./273.15 - 1./TABS) - &
+                    & 11.64*LOG(273.15/TABS) + 0.02265*(273.15 - TABS))
+               ELSE
+                  !SATURATION VAPOR PRESSURE WRT WATER (Bolton 1980)
+                  E1=SVP1*EXP(SVP2*(TABS-SVPT0)/(TABS-SVP3))
+               ENDIF
+                 QSFC_lnd(I)=EP2*E1/(PSFC(I)-ep_3*E1)             !specific humidity
+                 QSFC_lnd(I)=0.5*(QSFC_lnd(I) + QSFC(I))
+                 QSFCMR_lnd(I)=QSFC_lnd(I)/(1.-QSFC_lnd(I))       !mixing ratio
+              endif ! lsm
+            ENDIF
+            IF (icy(i)) THEN
+              if( lsm == lsm_ruc) then
+                QSFCMR_ice(I)=QSFC_ice(I)/(1.-QSFC_ice(I))       !mixing ratio 
+              else
+               IF (TSK_ice(I) .LT. 273.15) THEN
+                  !SATURATION VAPOR PRESSURE WRT ICE (SVP1=.6112; 10*mb)
+                  E1=SVP1*EXP(4648*(1./273.15 - 1./TSK_ice(I)) - &
+                    & 11.64*LOG(273.15/TSK_ice(I)) + 0.02265*(273.15 - TSK_ice(I)))
+               ELSE
+                  !SATURATION VAPOR PRESSURE WRT WATER (Bolton 1980)
+                  E1=SVP1*EXP(SVP2*(TSK_ice(I)-SVPT0)/(TSK_ice(i)-SVP3))
+               ENDIF
+                 QSFC_ice(I)=EP2*E1/(PSFC(I)-ep_3*E1)             !specific humidity
+                 QSFCMR_ice(I)=EP2*E1/(PSFC(I)-E1)                !mixing ratio
+              endif ! lsm
+            ENDIF
+
+         ELSE
+
+            ! Use what comes out of the LSM, NST, and CICE
+            IF (wet(i)) QSFCMR_wat(I)=QSFC_wat(I)/(1.-QSFC_wat(I))
+            IF (dry(i)) QSFCMR_lnd(I)=QSFC_lnd(I)/(1.-QSFC_lnd(I))
+            IF (icy(i)) QSFCMR_ice(I)=QSFC_ice(I)/(1.-QSFC_ice(I))
+
+         ENDIF
+      ENDDO
+
       IF (debug_code >= 1) THEN
         write(0,*)"ITIMESTEP=",ITIMESTEP," iter=",iter
         DO I=its,ite
            write(0,*)"=== important input to mynnsfclayer, i:", i
            IF (dry(i)) THEN
-             write(0,*)"dry=",dry(i)," pblh=",pblh(i)," tsk=", tskin_lnd(i),&
+             write(0,*)"dry=",dry(i)," =",pblh(i)," tsk=", tskin_lnd(i),&
```

Comment:
Lets keep the "pblh"

Reply:
@joeolson42 Sorry, Joe, I removed it by accident. Just committed the fix to that.